openml-utils/src/main/java/com/feedzai/openml/util/data/ClassificationDatasetSchemaUtil.java

```diff
@@ -43,16 +44,13 @@ private ClassificationDatasetSchemaUtil() { }
      * problem.
      *
      * @param datasetSchema The {@link DatasetSchema}.
-     * @return The number of classes on the target variable.
+     * @return The number of classes on the target variable, if a target variable is defined, or {@link Optional#empty()} otherwise.
      */
-    public static int getNumClassValues(final DatasetSchema datasetSchema) {
-
-        final AbstractValueSchema valueSchema = datasetSchema
-                .getFieldSchemas()
-                .get(datasetSchema.getTargetIndex())
-                .getValueSchema();
-
-        return getNumClassValues(valueSchema);
+    public static Optional<Integer> getNumClassValues(final DatasetSchema datasetSchema) {
```

Comment:
this is a breaking change in the openml-utils module. Since we are still on version 0.X, a new minor should be enough, but it can't be a new hotfix

Reply:
Thanks for the warning!